crates/sozo/voyager/src/analyzer.rs

```diff
@@ -0,0 +1,765 @@
+//! Project analyzer for extracting Dojo project information
+
+use std::fs;
+use std::path::{Path, PathBuf};
+
+use anyhow::{anyhow, Result};
+use serde_json;
+use starknet_crypto::Felt;
+use tracing::debug;
+
+use crate::config::{ArtifactType, ContractArtifact, FileInfo, Manifest, StarknetArtifacts};
+
+/// Project analyzer for extracting Dojo project information
+#[derive(Debug)]
+pub struct ProjectAnalyzer {
+    project_root: PathBuf,
+}
+
+impl ProjectAnalyzer {
+    pub fn new(project_root: PathBuf) -> Self {
+        Self { project_root }
+    }
+
+    /// Extract Dojo version from Scarb.toml
+    pub fn extract_dojo_version(&self) -> Option<String> {
+        let scarb_toml_path = self.project_root.join("Scarb.toml");
+
+        let contents = fs::read_to_string(&scarb_toml_path).ok()?;
+        let parsed: toml::Value = toml::from_str(&contents).ok()?;
+
+        // Look for dependencies.dojo.tag
+        parsed.get("dependencies")?.get("dojo")?.get("tag")?.as_str().map(|s| s.to_string())
+    }
+
+    /// Extract package name from Scarb.toml
+    pub fn extract_package_name(&self) -> Result<String> {
+        let scarb_toml_path = self.project_root.join("Scarb.toml");
+
+        let contents = fs::read_to_string(&scarb_toml_path)
+            .map_err(|e| anyhow!("Failed to read Scarb.toml: {}", e))?;
+        let parsed: toml::Value =
+            toml::from_str(&contents).map_err(|e| anyhow!("Failed to parse Scarb.toml: {}", e))?;
+
+        // Look for package.name
+        parsed
+            .get("package")
+            .and_then(|p| p.get("name"))
+            .and_then(|n| n.as_str())
+            .map(|s| s.to_string())
+            .ok_or_else(|| anyhow!("No package name found in Scarb.toml"))
+    }
+
+    /// Extract license from Scarb.toml
+    pub fn extract_license(&self) -> Option<String> {
+        let scarb_toml_path = self.project_root.join("Scarb.toml");
+
+        let contents = fs::read_to_string(&scarb_toml_path).ok()?;
+        let parsed: toml::Value = toml::from_str(&contents).ok()?;
+
+        // Look for package.license
+        parsed
+            .get("package")
+            .and_then(|p| p.get("license"))
+            .and_then(|l| l.as_str())
+            .map(|s| s.to_string())
+    }
+
+    /// Discover contract artifacts from manifest file
+    pub fn discover_contract_artifacts(&self) -> Result<Vec<ContractArtifact>> {
+        debug!(
+            "Discovering contract artifacts from manifest file in: {}",
+            self.project_root.display()
+        );
+
+        // Try to find the manifest file (usually manifest_dev.json for dev profile)
+        let manifest_path = self.find_manifest_file()?;
```

Comment:
The manifest is generated after the migration, to give the user the state of the dojo world onchain.

Using the manifest here, may cause issue if it doesn't exist (first migration), or if it is outdated or imported from other project).

You should use the [`WorldLocal`](https://github.com/dojoengine/dojo/blob/c876437d20b101e051656fa846288340b65612bc/crates/dojo/world/src/local/artifact_to_local.rs#L40) which is loaded from the artifacts. It also handles the profile (which is important since users may want to migrate with a specific profile, hence the artifacts may change).

The `WorldLocal` also contains all the classes already loaded in case you need them.